pyccel/codegen/compiling/basic.py

```diff
@@ -236,3 +251,11 @@ def __eq__(self, other):
 
     def __hash__(self):
         return hash(self.target)
+
+    @property
+    def no_target_file(self):
+        """
+        Indicates whether the file has a target.
+        Eg an interface for a library may not have a target
+        """
+        return self._no_target_file
```

Comment:
Given how this is used in practice (`if not self.no_target_file`), I think that the opposite state should be stored instead, which would be called `has_target_file` and would be `True` by default.

I find the current solution confusing because of the double negation...

Reply:
Fixed b5f62690. (Tests cancelled as they will need to be rerun in 2mins when #886 is merged)

Reply:
#886 is merged now